bundles/persistence/org.openhab.persistence.influxdb/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java

```diff
@@ -445,16 +458,22 @@ private Object stateToObject(State state) {
    */
   private String stateToString(State state) {
     String value;
-    if (state instanceof PercentType) {
-      value = ((PercentType) state).toBigDecimal().toString();
-    } else if (state instanceof DateTimeType) {
-      value = String.valueOf(((DateTimeType) state).getCalendar().getTime().getTime());
-    } else if (state instanceof DecimalType) {
-      value = ((DecimalType) state).toBigDecimal().toString();
+    if (state instanceof DecimalType) {
+    	value = ((DecimalType) state).toBigDecimal().toString();
+    } else if (state instanceof PercentType) {
+    	value = ((PercentType) state).toBigDecimal().toString();
     } else if (state instanceof OnOffType) {
-      value = ((OnOffType) state) == OnOffType.ON ? DIGITAL_VALUE_ON : DIGITAL_VALUE_OFF;
+    	value = ((OnOffType) state) == OnOffType.ON ? DIGITAL_VALUE_ON : DIGITAL_VALUE_OFF;
+    } else if (state instanceof OpenClosedType) {
+    	value = ((OpenClosedType) state) == OpenClosedType.OPEN ? DIGITAL_VALUE_OPEN : DIGITAL_VALUE_CLOSED;  
```

Comment:
because we used 0/1 for storing the OpenClosedType (line 440) we must also use it for querying. Use DIGITAL_VALUE_ON/OFF here.